controllers/common/base_job.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+)
+
+// NewBaseJob creates a new kubernetes job with the given arguments
+func NewBaseJob(objectMeta metav1.ObjectMeta, app string, imageSpec perfv1alpha1.ImageSpec,
+	podConfig perfv1alpha1.PodConfigurationSpec) *batchv1.Job {
+
+	backoffLimit := int32(0)
```

Comment:
This should be parameterizable: iperf3 requires some retries.

Reply:
I'd keep the parameter set for this function as small as possible. If we passed every parameter to this function than at the end of the day we should pass batchv1.Job, which would make the function pointless. See the pattern in fio/sysbench, after creating the "base" job, they modify the few parameters they need (e.g. args, volumes, volumemounts).

Reply:
I second @julio-santiesteban 's idea.